torchvision/datasets/dtd.py

```diff
@@ -0,0 +1,86 @@
+import os
+import pathlib
+from typing import Optional, Callable, Union
+
+import PIL.Image
+
+from .utils import verify_str_arg, download_and_extract_archive
+from .vision import VisionDataset
+
+
+class DTD(VisionDataset):
+    """`Describable Textures Dataset (DTD) <https://www.robots.ox.ac.uk/~vgg/data/dtd/>`_.
+
+    Args:
+        root (string): Root directory of the dataset.
+        split (string, optional): The dataset split, supports ``"train"`` (default), ``"val"``, or ``"test"``.
+        fold (string or int, optional): The dataset fold. Should be ``1 <= fold <= 10``. Defaults to ``1``.
+        transform (callable, optional): A function/transform that  takes in an PIL image and returns a transformed
+            version. E.g, ``transforms.RandomCrop``.
+        target_transform (callable, optional): A function/transform that takes in the target and transforms it.
+    """
+
+    _URL = "https://www.robots.ox.ac.uk/~vgg/data/dtd/download/dtd-r1.0.1.tar.gz"
+    _MD5 = "fff73e5086ae6bdbea199a49dfb8a4c1"
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "train",
+        fold: Union[str, int] = 1,
+        download: bool = True,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+    ) -> None:
+        self._split = verify_str_arg(split, "split", ("train", "val", "test"))
+        self._fold = verify_str_arg(str(fold), "fold", [str(i) for i in range(1, 11)])
+
+        super().__init__(root, transform=transform, target_transform=target_transform)
+        self._base_folder = pathlib.Path(self.root) / type(self).__name__.lower()
+        self._data_folder = self._base_folder / "dtd"
+        self._meta_folder = self._data_folder / "labels"
+        self._images_folder = self._data_folder / "images"
+
+        if download:
+            self._download()
+
+        if not self._check_exists():
+            raise RuntimeError("Dataset not found. You can use download=True to download it")
+
+        self._image_files = []
+        classes = []
+        with open(self._meta_folder / f"{self._split}{self._fold}.txt") as file:
+            for line in file:
+                cls, name = line.strip().split("/")
+                self._image_files.append(self._images_folder.joinpath(cls, name))
+                classes.append(cls)
+
+        self.classes = sorted(set(classes))
+        self.class_to_idx = dict(zip(self.classes, range(len(self.classes))))
```

Comment:
Do you know why some of our datasets expose `self.classes` while others expose `self.labels`?

Also, what's the reasons for making these public? Should we document them?

Reply:
> Do you know why some of our datasets expose `self.classes` while others expose `self.labels`?

They are two different things: `self.classes` is a list of all class names whereas `self.labels` is a list of integers for every sample. Almost every dataset should have something similar to `self.labels`, while `self.classes` is not always available depending on the nature of the dataset.

> Also, what's the reasons for making these public?

Force of habit I guess. Most datasets that have these attributes also have them public. 

> Should we document them?

Given that these are the last additions to a (hopefully) soon to be legacy API, I wouldn't sweat it. For most other datasets they are also undocumented.